intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/trace/VirtualFileInProject.xtend

```diff
@@ -10,12 +10,19 @@ package org.eclipse.xtext.idea.trace
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VirtualFile
 import org.eclipse.xtend.lib.annotations.Data
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiUtil
 
 /**
  * @author Sebastian Zarnekow - Initial contribution and API
  */
 @Data
 class VirtualFileInProject {
+	
+	def static VirtualFileInProject forPsiElement(PsiElement element) {
+		return new VirtualFileInProject(PsiUtil.getVirtualFile(element), element.project)
```

Comment:
Can we use PsiUtil.getVirtualFile(element) at other locations, too, where we try to find the vfile for a psi element? e.g.
org.eclipse.xtext.idea.trace.TraceForVirtualFileProvider.tryHardToFindVirtualFile(PsiFile)
org.eclipse.xtext.psi.impl.BaseXtextFile.findVirtualFile(PsiFile)